actors/common/src/test/java/org/sunbird/BaseActorTest.java

```diff
@@ -0,0 +1,170 @@
+
+    when(InterServiceCommunicationFactory.getInstance())
+        .thenReturn(interServiceCommunication)
+        .thenReturn(interServiceCommunication);
+
+    PowerMockito.mockStatic(SystemSettingClientImpl.class);
+    SystemSettingClientImpl systemSettingClient = mock(SystemSettingClientImpl.class);
+    when(SystemSettingClientImpl.getInstance()).thenReturn(systemSettingClient);
+    when(systemSettingClient.getSystemSettingByFieldAndKey(
+            Mockito.any(ActorRef.class),
+            Mockito.anyString(),
+            Mockito.anyString(),
+            Mockito.anyObject()))
+        .thenReturn(new HashMap<>());
+
+    cassandraMock();
+    elasticSearchMock();
+  }
+
+  protected void interServiceCommunication() {
+    when(interServiceCommunication.getResponse(
+            Mockito.any(ActorRef.class), Mockito.any(Request.class)))
+        .thenReturn(getFirstEsInterServiceCommunicationResponse())
+        .thenReturn(getSecondEsInterServiceCommunicationResponse());
+  }
+
+  private void elasticSearchMock() {
+
+    when(ElasticSearchUtil.complexSearch(
+            Mockito.any(SearchDTO.class),
+            Mockito.eq(ProjectUtil.EsIndex.sunbird.getIndexName()),
+            Mockito.anyVararg()))
+        .thenReturn(null);
+
+    when(ElasticSearchUtil.getDataByIdentifier(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getEsDataByIdentifierResponse());
+  }
+
+  private void cassandraMock() {
+
+    when(cassandraOperation.updateRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(getSuccessResponse());
+
+    when(cassandraOperation.getAllRecords(Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getAllRecordResponse());
+
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(getSuccessResponse());
+
+    when(cassandraOperation.getRecordById(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getRecordByIdResponse());
+
+    when(cassandraOperation.getRecordById(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.anyList()))
+        .thenReturn(getCassandraRecordByIdForBulkUploadResponse());
+  }
+
+  private static Response getSuccessResponse() {
+    Response response = new Response();
+    response.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
+    return response;
+  }
+
+  private static Response getAllRecordResponse() {
+
+    Response response = new Response();
+    List<Map<String, Object>> resMapList = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    resMapList.add(map);
+    response.put(JsonKey.RESPONSE, resMapList);
+    return response;
+  }
+
+  protected abstract Map<String, Object> getEsDataByIdentifierResponse();
+
+  protected abstract Response getCassandraRecordByIdForBulkUploadResponse();
```

Comment:
return List<Response> = so that multiple thenReturn can be achieved by user exetrnally passing the list